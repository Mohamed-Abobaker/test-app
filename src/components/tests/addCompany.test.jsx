import React from 'react';
import { shallow } from 'enzyme';
import AddCompany from '../addCompany';

describe('AddCompany.jsx tests', () => {
  it('should render the correct component', () => {
    const wrapper = shallow(<AddCompany />)

    expect(wrapper.debug()).toMatchSnapshot();
  });
  it('should call correct function when cancel button is clicked', () => {
    const mockFunc = jest.fn();

    const wrapper = shallow(<AddCompany updateIsFormOpen={mockFunc} />);
    const buttons = wrapper.find('button');
    buttons.at(1).props().onClick()

    expect(mockFunc).toHaveBeenCalled();
  });
  it('should call correct function when form is submitted', () => {
    const mockFunc = jest.fn();
    const mockFunc1 = jest.fn();
    const updateFromMock = jest.fn();
    jest.spyOn(React, 'useState').mockReturnValue([{ companyName: 'name', sector: 'it', creationDay: '01-01-2001', country: 'UK', customers: 0, active: false },
      updateFromMock]);

    const wrapper = shallow(<AddCompany updateCompanies={mockFunc} companies={[{}]} updateIsFormOpen={mockFunc1} />);
    const forms = wrapper.find('form');
    forms.at(0).props().onSubmit({ preventDefault: () => { } })

    expect(mockFunc).toHaveBeenCalledWith([{}, {
      "active": false, "companyName": "name", "country": "UK", "creationDay":
        "01-01-2001", "customers": 0, "sector": "it"
    }]);
    expect(mockFunc1).toHaveBeenCalledWith(false)
  });
});