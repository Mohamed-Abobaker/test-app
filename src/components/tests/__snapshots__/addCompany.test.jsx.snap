// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCompany.jsx tests should render the correct component 1`] = `
"<div>
  <form onSubmit={[Function: onSubmit]}>
    <div>
      <label>
        Name
        :
        <input required={true} type=\\"text\\" value=\\"\\" onChange={[Function: onChange]} />
      </label>
    </div>
    <div>
      <label>
        Country
        :
        <input required={true} type=\\"text\\" value=\\"\\" onChange={[Function: onChange]} />
      </label>
    </div>
    <div>
      <label>
        Creation date
        :
        <input required={true} type=\\"date\\" value=\\"\\" onChange={[Function: onChange]} />
      </label>
    </div>
    <div>
      <label>
        sector
        :
        <input required={true} type=\\"text\\" value=\\"\\" onChange={[Function: onChange]} />
      </label>
    </div>
    <label>
      Number of customers:
      <input required={true} type=\\"number\\" min={0} value={0} onChange={[Function: onChange]} />
    </label>
    <label>
      Active:
      <select value={false} onChange={[Function: onChange]}>
        <option value={true}>
          true
        </option>
        <option value={false}>
          false
        </option>
      </select>
    </label>
    <div>
      <button type=\\"submit\\">
        Add
      </button>
      <button onClick={[Function: onClick]}>
        Cancel
      </button>
    </div>
  </form>
</div>"
`;
